lib/fswatcher/fswatcher.go

```diff
@@ -108,14 +119,24 @@ func (watcher *FsWatcher) watchFilesystem() {
 	}
 }
 
-func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
-	if isSubpath(eventPath, watcher.folderPath) {
+func (watcher *FsWatcher) newFsEvent(eventPath string) bool {
+	if len(watcher.fsEvents) >= maxFiles {
+		watcher.debugf("Tracking too many events, aggregating: %s\n", eventPath)
+	} else if _, ok := watcher.fsEvents["."]; ok {
+		watcher.debugf("Will scan entire folder anyway, dropping: %s\n", eventPath)
+	} else if isSubpath(eventPath, watcher.folderPath) {
 		path, _ := filepath.Rel(watcher.folderPath, eventPath)
-		if !watcher.shouldIgnore(path) {
-			return &FsEvent{path}
+		if watcher.pathInProgress(path) {
+			watcher.debugf("Skipping notification for finished path: %s\n",	path)
+		} else if watcher.shouldIgnore(path) {
+			watcher.debugf("Ignoring: %s\n", path)
+		} else {
+			return watcher.aggregateEvent(path, time.Now())
 		}
+	} else {
+		watcher.debugf("Bug: Detected change outside of folder, droping: %s\n", eventPath)
```

Comment:
On second thought I am not entirely sure reaching this really is a bug. Probably the notify backend does not follow symlinks and/or bind-mounts when recursively installing watches, but if it did this would be a legitimate scenario, not a bug.